metis-indexing/src/test/java/eu/europeana/indexing/IndexerPreprocessorTest.java

```diff
@@ -106,6 +108,78 @@ void preprocessRecordTierCalculationAndInitialiseExisting() throws Serialization
+        true,
+        List.of(), true, TierCalculationMode.INITIALISE, EnumSet.allOf(EdmType.class));
+
+    // when
+    TierResults results = IndexerPreprocessor.preprocessRecord(inputRdf, indexingProperties);
+
+    // then
+    List<String> tierProvidedData = extractTierDataByTarget(inputRdf.getAggregationList(),
+        Aggregation::getHasQualityAnnotationList, "/aggregation/provider/2022502/_KAMRA_356338");
+
+    List<String> tierEuropeanaData = extractTierDataByTarget(inputRdf.getEuropeanaAggregationList(),
+        EuropeanaAggregationType::getHasQualityAnnotationList,"/aggregation/europeana/2022502/_KAMRA_356338");
+
+    // verify two different aggregation has different calculations
+    assertArrayEquals(new String[]{CONTENT_TIER_BASE_URI + "3", METADATA_TIER_BASE_URI + "A"}, tierProvidedData.toArray());
+    assertArrayEquals(new String[]{CONTENT_TIER_BASE_URI + "3", METADATA_TIER_BASE_URI + "A"}, tierEuropeanaData.toArray());
+
+    // verify return of tier calculation
+    assertEquals("2", results.getMediaTier().toString());
+    assertEquals("A", results.getMetadataTier().toString());
+
+//    // verify return is equal to aggregation and not europeana aggregation
+//    assertTrue(tierProvidedData.contains(CONTENT_TIER_BASE_URI + results.getMediaTier().toString()) &&
+//        tierProvidedData.contains(METADATA_TIER_BASE_URI + results.getMetadataTier().toString()));
+//    assertFalse(tierEuropeanaData.contains(CONTENT_TIER_BASE_URI + results.getMediaTier().toString()) &&
+//        tierEuropeanaData.contains(METADATA_TIER_BASE_URI + results.getMetadataTier().toString()));
+  }
+
+  @Test
+  void preprocessRecordTierCalculationByTargetAndOverWriteExisting() throws SerializationException, IndexingException {
+    // given
+    final RdfConversionUtils conversionUtils = new RdfConversionUtils();
+    final RDF inputRdf = conversionUtils.convertStringToRdf(
+        IndexingTestUtils.getResourceFileContent("europeana_record_rdf_tier_for_overwrite.xml"));
+
+    final IndexingProperties indexingProperties = new IndexingProperties(Date.from(Instant.now()),
+        true,
+        List.of(), true, TierCalculationMode.OVERWRITE, EnumSet.allOf(EdmType.class));
+
+    // when
+    TierResults results = IndexerPreprocessor.preprocessRecord(inputRdf, indexingProperties);
+
+    // then
+    List<String> tierProvidedData = extractTierDataByTarget(inputRdf.getAggregationList(),
+        Aggregation::getHasQualityAnnotationList, "/aggregation/aggregator/2022502/_KAMRA_356338");
+
+    List<String> tierEuropeanaData = extractTierDataByTarget(inputRdf.getEuropeanaAggregationList(),
+        EuropeanaAggregationType::getHasQualityAnnotationList,"/aggregation/europeana/2022502/_KAMRA_356338");
+
+    // verify two different aggregation has different calculations
+    assertArrayEquals(new String[]{CONTENT_TIER_BASE_URI + "2", METADATA_TIER_BASE_URI + "B"}, tierProvidedData.toArray());
```

Comment:
? Do you mean CONTENT_TIER_2.getUri(); ??

Reply:
done.